R/gui.R

```diff
@@ -0,0 +1,74 @@
+parseCategoriesToRemove <- function(exclude.categories, all.dimnames)
+{
+    if (!nzchar(exclude.categories))
+        return(NULL)
+    # Handle edge case where user wants to remove one label and it contains a comma
+    trimmed.input <- trimws(exclude.categories)
+    if (length(trimmed.input) == 1L && trimmed.input %in% all.dimnames)
+        return(trimmed.input)
+    # Otherwise split the string using either ; or , depending on the input
+    sep <- ifelse(grepl(";", exclude.categories), ";", ",")
+    trimws(strsplit(exclude.categories, sep)[[1L]])
+}
+
+#' @title Parse Categories to Remove in GUI textboxes
+#' @param input.str A list containing two characters with the desired categories
+#'   in each dimension to be removed. The first element for the rows, the second for
+#'   the columns. The expected format is either a comma separated string or ;
+#'   separated string for each dimension.
+#' @param inputs A list containing all the inputs that have rows and columns that
+#'   are potentially removed.
+#' @return A list with two character vectors that contain the category names of
+#'   all row/column slices that are to be removed.
+#' @export
+ParseCategoriesToRemove <- function(input.str, inputs)
+{
+    stopifnot("List of length two required" =  length(input.str) == 2 && is.list(input.str),
+              "Strings expected in each list element" = all(vapply(input.str, is.character, logical(1L))))
+    inputs <- checkInputsAtMost2DOrQTable(inputs)
+    all.row.names <- unique(unlist(lapply(inputs, rowNames)))
+    all.col.names <- unique(unlist(lapply(inputs, colNames)))
+    all.dimnames <- list(all.row.names, all.col.names)
+    categories.to.remove <- mapply(parseCategoriesToRemove,
```

Comment:
need to assign names `c("row", "columns")` to `categories.to.remove` or not try to use the names in the Standard R code (referring to the rscript attached to the JIRA).